test/js/bun/http/form-data-set-append.test.js

```diff
@@ -0,0 +1,65 @@
+import { expect, test } from "bun:test";
+
+// https://github.com/oven-sh/bun/issues/12325
+
+test("formdata set with File works as expected", async () => {
+  const expected = ["617580375", "text-notes1.txt"];
+
+  const server = Bun.serve({
```

Comment:
nit: we can do `using` and that will automatically close the server without needing to manually call ``.stop`
```suggestion
  using server = Bun.serve({
```